File: LibVt6530/LibVt6530/PageProtocol.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LibVt6530
{
	public interface PageProtocol
	{
		void WriteBuffer(Page page, byte[] text);
		void WriteCursor(Page page, byte[] text);
		void WriteBuffer(Page page, string text);
		void WriteCursor(Page page, string text);
		void WriteChar(Page page, Cursor cursor, int c);
		void InsertChar(Page page);
		void DeleteChar(Page page);
		void Backspace(Page page);
		void Tab(Page page, int inc);
		void CarageReturn(Page page);
		void Linefeed(Page page);
		void Home(Page page);
		void End(Page page);
		
		void ValidateCursorPos(Page page);
		void SetCursor(Page page, int row, int col);
		void CursorLeft(Page page);
		void CursorRight(Page page);
		void CursorDown(Page page);
		void CursorUp(Page page);

		void ClearToEOL(Page page);
		void ClearToEOP(Page page);
		void ClearBlock(Page page, int startRow, int startCol, int endRow, int endCol);
		
		void ReadBuffer(StringBuilder outp, Page page, int reqMask, int forbidMask, int startRow, int startCol, int endRow, int endCol);
	};
}

File: Java/src/main/java/gov/revenue/vt6530/ui/PageProtocol.java
package gov.revenue.vt6530.ui;

import java.awt.Graphics;

/**
 *  Interface to the shared, unprotect, and
 *  protect mode virtual character display
 *  modes.
 */
public interface PageProtocol extends Attributes
{
	static final int MODE_DISPLAY = 0;
	static final int MODE_PROTECT = 1;
	static final int MODE_UNPROTECT = 2;

	void writeBuffer(Page page, String text);
	void writeCursor(Page page, String text);
	void writeChar(Page page, Cursor cursor, int c);
	
	void insertChar(Page page);
	void deleteChar(Page page);
	void backspace(Page page);
	void tab(Page page, int inc);
	void carageReturn(Page page);
	void linefeed(Page page);
	
	void validateCursorPos(Page page);
	void setCursor(Page page, int row, int col);
	void cursorLeft(Page page);
	void cursorRight(Page page);
	void cursorDown(Page page);
	void cursorUp(Page page);

	void home(Page page);
	void end(Page page);
	
	void clearToEOL(Page page);
	void clearToEOP(Page page);
	void clearBlock(Page page, int startRow, int startCol, int endRow, int endCol);
	
	byte[] readBuffer(Page page, int reqMask, int forbidMask, int startRow, int startCol, int endRow, int endCol);
}
